main/ogn-aprs.go

```diff
@@ -0,0 +1,240 @@
+/*
+	Copyright (c) 2022 Quentin Bossard
+	Distributable under the terms of The "BSD New" License
+	that can be found in the LICENSE file, herein included
+	as part of this header.
+
+	aprs.go: Routines for reading traffic from aprs
+*/
+
+package main
+
+import (
+	"bufio"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var aprsOutgoingMsgChan chan string = make(chan string, 100)
+var aprsIncomingMsgChan chan string = make(chan string, 100)
+var aprsExitChan chan bool = make(chan bool, 1)
+
+
+func authenticate(c net.Conn) {
+	filter := ""
+	if mySituation.GPSFixQuality > 0 {
+		filter = fmt.Sprintf(
+			"filter r/%.7f/%.7f/%d\r\n", mySituation.GPSLatitude, mySituation.GPSLongitude, 100)
+	}
+	auth := fmt.Sprintf("user OGNNOCALL pass -1 vers stratux 0.28 %s\r\n", filter)
+	fmt.Printf(auth)
+	fmt.Fprintf(c, auth)
+}
+
+func keepalive(c net.Conn) {
+	ticker := time.NewTicker(240 * time.Second)
+
+	go func() {
+		for t := range ticker.C {
+			fmt.Fprintf(c, "# stratux keepalive %s\n", t)
+			fmt.Printf("# stratux keepalive %s\n", t)
+		}
+	}()
+}
+
+func updateFilter(c net.Conn) {
+	ticker := time.NewTicker(10 * time.Second)
+	go func() {
+		for range ticker.C {
+			if mySituation.GPSFixQuality > 0 {
+				filter := fmt.Sprintf(
+					"#filter r/%.7f/%.7f/%d\r\n", mySituation.GPSLatitude, mySituation.GPSLongitude, 100)
+				// filter = "#filter?\r\n"
+				fmt.Fprintf(c, filter)
+			}
+		}
+	}()
+}
+
+func aprsListen() {
+	// https://regex101.com/r/Cv9mSq/1
+	// rex := regexp.MustCompile(`(ICA|FLR|SKY|PAW|FNT)([\dA-Z]{6})>[A-Z]+,qAS,([\d\w]+):\/(\d{6})h(\d*\.?\d*[NS])[\/\\](\d*\.?\d*[EW])['\^nX](\d{3})\/(\d{3})\/A=\d*\s!W(\d+)!\sid([\dA-F]{8})`)
+	rex := regexp.MustCompile(
+		`(ICA|FLR|SKY|PAW|OGN|RND|FMT|MTK|XCG|FAN|FNT)([\dA-Z]{6})>` + // protocol, id
```

Comment:
Guess I will just believe you with that one ;-)
Might be more readable with named capture groups.. but I guess the comments solve the same purpose

Reply:
i avoided names capture groups for performances purposes. if you think it's an over-optimization, i'm happy to add them to the regexp